### PR TITLE
Don't prefix empty lines.

### PR DESCRIPTION
**Pull request by mek-apelsin**
Don't prefix with whitespaces when the added line is empty.

### CODE REVIEW

File: WhitespacePreservation.md [hunk 1]
```diff
@@ -50,18 +50,22 @@ for filename, codeblock := range files {
...
+empty.
 
 ```go "Handle replace line"
 matches := replaceRe.FindStringSubmatch(line)
 if matches == nil {
-	ret += CodeBlock(prefix) + CodeBlock(line)
+	if line != "\n" {
+		ret += CodeBlock(prefix)
+	}
+	ret += CodeBlock(line)
```
Comment (mek-apelsin) [inline WhitespacePreservation.md]: The lines 65-68 might be clearer if they are 
```go
switch line {
case "\n":
	ret += CodeBlock(line)
default:
	ret += CodeBlock(prefix) + CodeBlock(line)
}
```

Comment (driusan) [reply WhitespacePreservation.md]: Feel free to change it if you like it better.

**Comment by driusan** [comment]
I'm not sure if this is a safe thing to do with all whitespace sensitive languages, but I can't think of any counter-examples, so this should probably be fine.

File: WhitespacePreservation.md (hunk 1 above)
Comment (mek-apelsin) [reply WhitespacePreservation.md]: Not really, I liked the first version better. But I am still unsure what is best if you go back and read it later.

**Comment by mek-apelsin** [comment]
I actually implemented this to be compliant with PEP8 (python), GNU make doesn't complain, and as a added bonus go fmt started to accept the result without "fixing" it. The testing I have done is obviously not great, but without learning Haskell or digging through other make implementations, I am unsure what to do.

**Comment by driusan** [comment]
I'm pretty sure Haskell allows blank lines, I'll just merge this and if necessary we can always revert it or add a flag if we discover some other language needs whitespace-only lines within blocks.

**Pull request merged**